src/stores/filings.ts

```diff
@@ -75,16 +74,21 @@ export const useBcrosFilings = defineStore('bcros/filings', () => {
   }
 
   const loadFilings = async (identifier: string, force = false) => {
-    const businessCached = filings.value && identifier === _filingsForIdentifier.value
+    const businessCached = (filings.value.length > 0)
     if (!businessCached || force) {
-      filings.value = await getFilings(identifier)
+      const newFilings = await getFilings(identifier)
+      filings.value.push(...newFilings)
     }
   }
 
   const clearFilings = () => {
     filings.value = []
   }
 
+  const insertFiling = (filing: ApiResponseFilingI) => {
+    filings.value.unshift(filing)
+  }
```

Comment:
Note: whether insertFiling is called before the filings are fetched (list is empty) or after the filings are fetched, it will insert the filing at the start (top) of the list, keeping a potential pending task at the top.